scripts/upload_sample.py

```diff
@@ -0,0 +1,196 @@
+    json_res = resp.json()
+    token_obj = TokenObject(token=json_res["access_token"], type=json_res["token_type"])
+    return token_obj
+
+
+def api_authentication(func: Callable) -> Callable:
+    """Use authentication token for api.
+
+    :param func: API function to wrap with API auth headers
+    :type func: Callable
+    :return: Wrapped API function
+    :rtype: Callable
+    """
+
+    @wraps(func)
+    def wrapper(token_obj: TokenObject, *args, **kwargs) -> Callable:
+        """Add authentication headers to API requests.
+
+        :param token_obj: Auth token object
+        :type token_obj: TokenObject
+        :return: Wrapped API call function
+        :rtype: Callable
+        """
+        headers = CaseInsensitiveDict()
+        headers["Accept"] = "application/json"
+        headers["Authorization"] = f"{token_obj.type.capitalize()} {token_obj.token}"
+
+        return func(headers=headers, *args, **kwargs)
+
+    return wrapper
+
+
+@api_authentication
+def upload_sample(headers: CaseInsensitiveDict, ctx: ExecutionContext, cnf: SampleConfig) -> str:
+    """Create a new sample."""
+    sample_obj = json.load(cnf.prp_result.open())
+    resp = requests.post(f'{ctx.api_url}/samples', headers=headers, json=sample_obj, timeout=TIMEOUT)
+
+    resp.raise_for_status()
+    resp_data = resp.json()
+    return resp_data["sample_id"]
+
+
+@api_authentication
+def upload_signature(headers: CaseInsensitiveDict, ctx: ExecutionContext, cnf: SampleConfig, sample_id: str) -> str:
+    """Upload a genome signature to sample."""
+    resp = requests.post(f'{ctx.api_url}/samples/{sample_id}/signature', headers=headers, files={"signature": cnf.minhash_signature.open()}, timeout=TIMEOUT)
+
+    resp.raise_for_status()
+    return resp.json()
+
+
+@api_authentication
+def add_sample_to_group(headers: CaseInsensitiveDict, ctx: ExecutionContext, cnf: SampleConfig, sample_id: str) -> str:
+    """Add sample to a group."""
+    resp = requests.put(f'{ctx.api_url}/groups/{cnf.group_id}/sample', headers=headers, params={"sample_id": sample_id}, timeout=TIMEOUT)
+
+    resp.raise_for_status()
+    return resp.json()
+
+
+def _process_generic_status_codes(error, sample_id):
+    """Process generic http status codes."""
+    is_major_error = True
+    match error.response.status_code:
+        case 404:
+            msg = f"Sample with {sample_id} is not in Bonsai"
+            is_major_error = False
+        case 500:
+            msg = "An unexpected error occured in Bonsai, check bonsai api logs"
+        case _:
+            msg = f"An unknown error occurred; {str(error)}"
+    return msg, is_major_error
+
+
+@click.command()
+@click.option("-a", "--api", type=str, help="Upload configuration")
+@click.option("-u", "--user", envvar=USER_ENV, type=str, help="Username")
+@click.option("-p", "--password", envvar=PASSWD_ENV, type=str, help="Password")
+@click.option("-c", "--config", required=True, type=click.File(), help="Upload configuration")
```

Comment:
I think this is the wrong name to call it. It's the input yaml. I suggest `-i` and `--input` or `--input_yaml`